ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/Main.scala

```diff
@@ -49,232 +35,14 @@ object Main extends StrictLogging {
-        jsonApiDbConfig <- Future.successful(config(dbInstance, dbConfig))
-        a <- fn(Some(jsonApiDbConfig))
-        _ <- Future.successful(
-          stop(dbInstance, dbConfig) // XXX ignores resulting Try
-        ) // TODO: use something like `lf.data.TryOps.Bracket.bracket`
-      } yield a
-
-    case None => fn(None)
-  }
-
-  private[this] final case class QueryStoreBracket[S, D](
-      state: () => S,
-      start: S => Try[D],
-      config: (S, D) => JdbcConfig,
-      stop: (S, D) => Try[Unit],
-  )
-  private[this] object QueryStoreBracket {
-    type T = QueryStoreBracket[_, _]
-    val Postgres: T = QueryStoreBracket[PostgresRunner, PostgresDatabase](
-      () => new PostgresRunner(),
-      _.start(),
-      (_, d) => jsonApiJdbcConfig(d),
-      (r, _) => r.stop(),
-    )
-
-    import com.daml.testing.oracle, oracle.OracleAround
-    val Oracle: T = QueryStoreBracket[OracleRunner, OracleAround.RichOracleUser](
-      () => new OracleRunner,
-      _.start(),
-      _ jdbcConfig _,
-      _.stop(_),
-    )
-
-    private[this] final class OracleRunner {
-
-      private val defaultUser = "ORACLE_USER"
-      private val retainData = sys.env.get("RETAIN_DATA").exists(_ equalsIgnoreCase "true")
-      private val useDefaultUser = sys.env.get("USE_DEFAULT_USER").exists(_ equalsIgnoreCase "true")
-      type St = OracleAround.RichOracleUser
-
-      def start(): Try[St] = Try {
-        if (useDefaultUser) OracleAround.createOrReuseUser(defaultUser)
-        else OracleAround.createNewUniqueRandomUser()
-      }
-
-      def jdbcConfig(user: St): JdbcConfig = {
-        import DbStartupMode._
-        val startupMode: DbStartupMode = if (retainData) CreateIfNeededAndStart else CreateAndStart
-        JdbcConfig(
-          dbutils.JdbcConfig(
-            "oracle.jdbc.OracleDriver",
-            user.jdbcUrlWithoutCredentials,
-            user.oracleUser.name,
-            user.oracleUser.pwd,
-            ConnectionPool.PoolSize.Production,
-          ),
-          startMode = startupMode,
-        )
-      }
-
-      def stop(user: St): Try[Unit] = {
-        if (retainData) Success(()) else Try(user.drop())
-      }
-    }
-
-    def lookup(q: QueryStoreIndex): Option[T] = q match {
-      case QueryStoreIndex.No => None
-      case QueryStoreIndex.Postgres => Some(Postgres)
-      case QueryStoreIndex.Oracle => Some(Oracle)
-    }
-  }
-
-  private def jsonApiJdbcConfig(c: PostgresDatabase): JdbcConfig =
-    JdbcConfig(
-      dbutils
-        .JdbcConfig(
-          driver = "org.postgresql.Driver",
-          url = c.url,
-          user = "test",
-          password = "",
-          ConnectionPool.PoolSize.Production,
-        ),
-      startMode = DbStartupMode.CreateOnly,
-    )
-
-  private def resolveSimulationClass(str: String): Throwable \/ Class[_ <: Simulation] = {
-    try {
+  private def resolveSimulationClass(
+      str: String
+  )(implicit ec: ExecutionContext): Future[Class[_ <: Simulation]] = {
```

Comment:
Of course this existed already, is instantiation by reflection really necessary? Rather avoid doing that, if necessary pattern match on some config and create the actual class that is needed, which would be far more readable.